From ab96d4060b93610c07400e4e324cfc27a5475815 Mon Sep 17 00:00:00 2001
From: Adrien Ricciardi <aricciardi@baylibre.com>
Date: Fri, 6 Oct 2023 10:23:17 +0200
Subject: [PATCH 23/38] net: can: Added a 'private' field to store a driver
 internal structure.

Also forced some constants type to be unsigned to fix compilation warnings.

Signed-off-by: Adrien Ricciardi <aricciardi@baylibre.com>
---
 net/can/Can_GeneralTypes.h | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/net/can/Can_GeneralTypes.h b/net/can/Can_GeneralTypes.h
index 05174a7d..9390a735 100644
--- a/net/can/Can_GeneralTypes.h
+++ b/net/can/Can_GeneralTypes.h
@@ -30,13 +30,13 @@
 #include <ComStack_Types.h>
 #include <tpl_os.h>
 
-#define TPL_CAN_ID_TYPE_STANDARD (0x00 << 30)
-#define TPL_CAN_ID_TYPE_FD_STANDARD (0x01 << 30)
-#define TPL_CAN_ID_TYPE_EXTENDED (0x02 << 30)
-#define TPL_CAN_ID_TYPE_FD_EXTENDED (0x03 << 30)
-#define TPL_CAN_ID_TYPE_MASK (0x03 << 30)
+#define TPL_CAN_ID_TYPE_STANDARD (0x00U << 30)
+#define TPL_CAN_ID_TYPE_FD_STANDARD (0x01U << 30)
+#define TPL_CAN_ID_TYPE_EXTENDED (0x02U << 30)
+#define TPL_CAN_ID_TYPE_FD_EXTENDED (0x03U << 30)
+#define TPL_CAN_ID_TYPE_MASK (0x03U << 30)
 #define TPL_CAN_ID_TYPE_GET(id) ((id & TPL_CAN_ID_TYPE_MASK) >> 30)
-#define TPL_CAN_ID_MASK (0x3FFFFFFF)
+#define TPL_CAN_ID_MASK (0x3FFFFFFFU)
 
 #define TPL_CAN_CLASSIC_FRAME_MAXIMUM_PAYLOAD_SIZE (8)
 #define TPL_CAN_FD_FRAME_MAXIMUM_PAYLOAD_SIZE (64)
@@ -137,6 +137,7 @@ struct tpl_can_controller_t
 	Std_ReturnType (*transmit)(struct tpl_can_controller_t *ctrl, const Can_PduType *pdu_info);
 	Std_ReturnType (*receive)(struct tpl_can_controller_t *ctrl, Can_PduType *pdu_info);
 	int (*is_data_available)(struct tpl_can_controller_t *ctrl);
+	void *priv; // Store a custom structure for the driver internal use
 };
 typedef struct tpl_can_controller_t tpl_can_controller_t;
 
-- 
2.34.1

